swarm/src/behaviour.rs

```diff
@@ -141,7 +139,11 @@ pub trait NetworkBehaviour: 'static {
     ///
     /// Note that the handler is returned to the [`NetworkBehaviour`] on connection failure and
     /// connection closing.
-    fn new_handler(&mut self) -> Self::ConnectionHandler;
+    fn new_handler(
+        &mut self,
+        peer: &PeerId,
```

Comment:
```suggestion
        peer: PeerId,
```

Why provide a reference to a `Copy` type?